luigi/contrib/azureblob.py

```diff
@@ -74,7 +75,8 @@ def connection(self):
                                                  connection_string=self.kwargs.get("connection_string"),
                                                  endpoint_suffix=self.kwargs.get("endpoint_suffix"),
                                                  custom_domain=self.kwargs.get("custom_domain"),
-                                                 is_emulated=self.kwargs.get("is_emulated") or False)
+                                                 is_emulated=self.kwargs.get("is_emulated", False),
+                                                 token_credential=self.kwargs.get("token_credential"))
```

Comment:
will `azure-storage<=0.36` allow for the passing of `token_credential` without fail? (even if without impact)

Reply:
no, it immediately fails
```
>>> blockblobservice.BlockBlobService(account_name="a", token_credential="a")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'token_credential'
```
do you prefer to keep some compatibility in here in case users aren't able to upgrade their packages?

Reply:
Oh wait, i see the hard update on az dependencies anyway.

So either we make this change, as is, and include it as an explicit breaking change. Or we explore options for enabling both.

I'm not immediately thinking of an easy way to keep compatibility. 

azure-storage==0.36 is also from 2017...

Reply:
I can refactor these keyword arguments in a dict, and add `token_credential` argument only if we're using the new azure package:

```python
from pkg_resources import parse_version
new_azure_package = parse_version(azure.storage.blob.__version__) >= parse_version("2.1.0")
```

(Under a try/except AttributeError because this attribute didn't exist in the old package, it's `azure.storage.__version__` )

Reply:
I pushed it in another branch https://github.com/spotify/luigi/commit/0566938d6b36283378cbfe1f9c0004699c79f570
I can cherry-pick this here, it's up to you